app/ide-desktop/lib/dashboard/src/components/SelectionBrush.tsx

```diff
@@ -11,129 +11,148 @@ import * as eventModule from '#/utilities/event'
-          setPosition({ left: event.pageX, top: event.pageY })
-        }
+    }
+    const onMouseUp = (event: MouseEvent) => {
+      if (didMoveWhileDraggingRef.current) {
+        onDragEndRef.current(event)
       }
-      const onClick = (event: MouseEvent) => {
-        if (isMouseDownRef.current && didMoveWhileDraggingRef.current) {
-          event.stopImmediatePropagation()
-        }
+      // The `setTimeout` is required, otherwise the values are changed before the `onClick` handler
+      // is executed.
+      window.setTimeout(() => {
+        isMouseDownRef.current = false
+        didMoveWhileDraggingRef.current = false
+      })
+      setAnchor(null)
+    }
+    const onMouseMove = (event: MouseEvent) => {
+      if (!(event.buttons & 1)) {
+        isMouseDownRef.current = false
       }
-      const onDragStart = () => {
-        if (isMouseDownRef.current) {
-          isMouseDownRef.current = false
-          onDragCancelRef.current()
-          setAnchor(null)
-        }
+      if (isMouseDownRef.current) {
+        // Left click is being held.
+        didMoveWhileDraggingRef.current = true
+        lastMouseEvent.current = event
+        const positionLeft =
+          parentBounds.current == null
+            ? event.pageX
+            : Math.max(
+                parentBounds.current.left - margin,
+                Math.min(parentBounds.current.right + margin, event.pageX)
+              )
+        const positionTop =
+          parentBounds.current == null
+            ? event.pageY
+            : Math.max(
+                parentBounds.current.top - margin,
+                Math.min(parentBounds.current.bottom + margin, event.pageY)
+              )
+        setPosition({ left: positionLeft, top: positionTop })
       }
-
-      target.addEventListener('mousedown', onMouseDown)
-      document.addEventListener('mouseup', onMouseUp)
-      document.addEventListener('dragstart', onDragStart, { capture: true })
-      document.addEventListener('mousemove', onMouseMove)
-      document.addEventListener('click', onClick, { capture: true })
-
-      return () => {
-        target.removeEventListener('mousedown', onMouseDown)
-        document.removeEventListener('mouseup', onMouseUp)
-        document.removeEventListener('dragstart', onDragStart, { capture: true })
-        document.removeEventListener('mousemove', onMouseMove)
-        document.removeEventListener('click', onClick, { capture: true })
+    }
+    const onClick = (event: MouseEvent) => {
+      if (isMouseDownRef.current && didMoveWhileDraggingRef.current) {
+        event.stopImmediatePropagation()
       }
-    } else {
-      return () => {}
     }
-  }, [/* should never change */ modalRef, targetRef])
+    const onDragStart = () => {
+      if (isMouseDownRef.current) {
+        isMouseDownRef.current = false
+        onDragCancelRef.current()
+        setAnchor(null)
+      }
+    }
+
+    document.addEventListener('mousedown', onMouseDown)
```

Comment:
Can we return `target` here instead of `document`? this was intended - we want to start selection only on the particular element